install/scripts/craycli_init.py

```diff
@@ -0,0 +1,753 @@
+        keycloak_base = os.environ.get('KEYCLOAK_BASE', DEFAULT_KEYCLOAK_BASE)
+        ks = KeycloakSetup(
+            keycloak_base=keycloak_base,
+            kc_master_admin_client_id=kc_master_admin_secrets['client_id'],
+            kc_master_admin_username=kc_master_admin_secrets['user'],
+            kc_master_admin_password=kc_master_admin_secrets['password'],
+        )
+        ks.delete_user(tmpUser.tu_username)
+    else:
+        LOGGER.error("During cleanup no CLI User Auth Secret present")
+
+    # remove the secret from k8s
+    tmpUser.deleteSecret()
+
+    # remove the individual configuration files to 'uninitialize' cray cli
+    LOGGER.info("Uninitializing nodes:")
+    nodes = getSlsNodes(k8sClientApi)
+    cmdOpt = "--cleanupnode"
+    callRemoteFunc(nodes, cmdOpt)
+
+def doReinitCleanup(k8sClientApi, username, password):
+    # clean out the temporary user initialization
+    doCleanup(k8sClientApi)
+
+    LOGGER.info(f"Re-initializing the cray cli with existing keycloak user {username}")
+
+    # create a new secret with the new username and password - used
+    # by individual node initialization
+    reinitUser = CliUserAuth(
+            createIfNeeded = True,
+            k8sClientApi = k8sClientApi,
+            username = username,
+            password = password)
+    if reinitUser.tu_username != "":
+        # the initnode function pulls user/password from secret we just created
+        LOGGER.info("Initializing nodes:")
+        nodes = getSlsNodes(k8sClientApi)
+        cmdOpt = "--initnode"
+        callRemoteFunc(nodes, cmdOpt)
+    else:
+        LOGGER.error("Failed to create new reinitialization user secret")
+
+    # remove the secret from k8s
+    reinitUser.deleteSecret()
+
+
+def main():
+    # get the command line arguments
+    # NOTE: user must specify one of the following
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--run", action="store_true", 
+        help="Run the script to create keycloak user and initialize craycli on all ncn hosts")
+    parser.add_argument("--cleanup", action="store_true", 
+        help="Remove craycli initialization and clean up keycloak user")
+    parser.add_argument("--initnode", action="store_true", 
+        help="Initiailze craycli on this host")
+    parser.add_argument("--cleanupnode", action="store_true", 
+        help="Cleanup craycli on this host")
+    parser.add_argument("--debug", action="store_true", 
+        help="Display debug level log messages")
+    parser.add_argument('-u', "--username", nargs='?', 
+        help='Optional new user for re-init on cleanup')
+    parser.add_argument('-p', "--password", nargs='?', 
+        help='Optional password for re-init on cleanup')
+    parser.add_argument('-g', "--group", nargs='?',
+        help='Optional group for new keycloak user permissions')
+    args = parser.parse_args()
+
+    # set up logging
+    logLevel = logging.INFO
+    if args.debug == True:
```

Comment:
args.debug is a boolean flag, so you don't need to compare it to True, you can just use it:
if args.debug:
The same goes for any further comparisons below.

Reply:
Good catch.